widgets/WallEntry.php

```diff
@@ -46,27 +46,14 @@ class WallEntry extends \humhub\modules\content\widgets\WallEntry
         ]
     ];
 
-    public function getContextMenu()
+    public function getControlsMenuEntries()
     {
-        $canEdit = $this->contentObject->content->canEdit();
-        if($canEdit) {
-            $this->controlsOptions = [
-                'add' => [
-                    [CloseLink::class, ['entry' => $this->contentObject], ['sortOrder' => 210]]
-                ]
-            ];
-        }
+        $result = parent::getControlsMenuEntries();
 
-        if($this->stream) {
-            return parent::getContextMenu();
-        }
-
-        $this->controlsOptions['prevent'] = [\humhub\modules\content\widgets\EditLink::class , \humhub\modules\content\widgets\DeleteLink::class];
-        $result = parent::getContextMenu();
-
-        if($canEdit) {
-            $this->addControl($result, [EditLink::class, ['entry' => $this->contentObject], ['sortOrder' => 100]]);
-            $this->addControl($result, [DeleteLink::class, ['entry' => $this->contentObject], ['sortOrder' => 200]]);
+        if ($this->model->content->canEdit()) {
+            $result[] = [CloseLink::class, ['entry' => $this->model], ['sortOrder' => 210]];
+        } else {
+            $this->renderOptions->disableControlsEntryEdit()->disableControlsEntryDelete();
```

Comment:
This will only disable the default edit/delete controls links. We have to add the replacement widgets as in the old implementation if the user is allowed to edit.

Reply:
I didn't find a reason to keep the old code where menu entries `EditLink::class` and `DeleteLink::class` where added by condition `if($canEdit)` because the method `parent::getControlsMenuEntries()` already returns the entries, so make sense only to disable them by condition `if(!$canEdit)`.

Maybe the following code is needed only to set new `sortOrder` against what defined in the parent method by default:
```
if($canEdit) {
    $result[] = [EditLink::class, ['entry' => $this->model], ['sortOrder' => 100]];
    $result[] = [DeleteLink::class, ['entry' => $this->model], ['sortOrder' => 200]];
}
```
Is it true? If yes then we need to redefine this only for `DeleteLink` in order to change default `sortOrder` from `300` to `200`, but `EditLink` have same order `100` in both places, one difference is that in parent method we have `'url' => $this->getEditUrl()` but in current no such option.

Please correct me and explain if I am wrong, because I changed the code only after complete testing the code with `$canEdit` = `false` and `true`.

Reply:
Those are not the default widgets from the core but:

- https://github.com/humhub/humhub-modules-calendar/blob/master/widgets/EditLink.php
- https://github.com/humhub/humhub-modules-calendar/blob/master/widgets/DeleteLink.php

Reply:
done in commit https://github.com/humhub/humhub-modules-calendar/commit/45b339a206d70f60fab44a91c7013ad5e0dfe204